tsa-core/src/main/kotlin/org/usvm/machine/interpreter/TvmArtificialInstInterpreter.kt

```diff
@@ -155,11 +155,15 @@ class TvmArtificialInstInterpreter(
             return null
         }
         val (head, tail) =
-            stmt.sentMessages.splitHeadTail() ?: return null
+            stmt.sentMessages.splitHeadTail()
+                ?: return null
         val nextInst = stmt.copy(sentMessages = tail)
         val currentContractToPush = currentContract
         val pushArgsOnStack: TvmState.() -> Unit = {
+            val messageCounter = sentMessagesCounter[currentContractToPush] ?: 0
+            sentMessagesCounter = sentMessagesCounter.put(currentContractToPush, messageCounter + 1)
```

Comment:
I believe, the point of this counter was that it should be separate for each contract. For example, first message that contract A sent. I don't think that global counter is very useful. You can clarify details with @Karkarmath

Reply:
But not for each exection. Or is this fine?

Another thing with the counter (more serious): it doen't take into account `unprocessedMessages`

Reply:
It does take into account all the messages that were sent into the blockchain, even with potentially null receiver, as long as the message are handled in `on_out_message` handler. I'll ask @Karkarmath to give a better specification.

Reply:
Ok, then only question about different calls to one contract is left.